### DISCUSSION TITLE
Image is missing required "src" property ?

### DISCUSSION THREAD

**Discussion by gurupal** [vercel/next.js#18531] [Help]
Why I am getting this weird error while passing image data. Image "src" is already defined.

Uncaught Error: Image is missing required "src" property. Make sure you pass "src" in props to the `next/image` component. Received: {"width":300,"height":500}

`<div className={styles.image_container}>
                <Link href={`/p/${encodeURIComponent(product.slug)}`}>
                    <a>
                    <Image
                        src={product.displayImage}
                        alt={product.name}
                        width={300}
                        height={500}
                    />
                    </a>
                </Link>
            </div>`

**Comment by MathiasKandelborg** [comment]
It seems your `{product.displayImage}` value isn't set properly. Try `console.log(product.displayImage)` before that code, to see if the value exists.

**Reply by gurupal** [reply]
@MathiasKandelborg  i did console log .. and it is showing the image links for all the products

**Reply by MathiasKandelborg** [reply]
I'm not sure I understand; it should be a problem with that value `displayImage` though. There's not really any way around it, try and hardcode a string, and the error will most likely disappear.

**Reply by gurupal** [reply]
@MathiasKandelborg 
Can you check this, if there is something wrong with code. Else will search for more . thank you

![Screenshot 2020-10-30 at 4 41 54 PM](https://user-images.githubusercontent.com/5556766/97698669-ddb2cd80-1ace-11eb-8695-3438f5ed1157.png)

**Reply by MathiasKandelborg** [reply]
It might be a problem while hydrating. Try passing `product.displayImage || ''`

**Reply by mauricioross** [reply]
Hi! any solution for this? I have same problem

**Reply by zhyd1997** [reply]
@mauricioross 
Just try `<img>` tag and see what happens, if it's also not working, maybe there is an error at `product.displayImage` type.

**Reply by zhyd1997** [reply]
Or you can try this code: 
```jsx
<Img
    src={`${product.displayImage}`} // core
    ...other_codes_here
>
```

**Reply by gurupal** [reply]
but i want to use Image component of nextjs.. not img tag

**Reply by zhyd1997** [reply]
@gurupal 

Yes, it's recommend, but the 1st approach just for test if `product.displayImage` works,

Or just try the 2nd approach.

**Reply by gurupal** [reply]
yes ,,, i have used img tag which is working fine for me...

**Reply by zhyd1997** [reply]
Awesome, and how about the 2nd approach?

**Reply by gurupal** [reply]
with the Image component .. nope

**Reply by zhyd1997** [reply]
@gurupal 
I found related source code here: https://github.com/vercel/next.js/blob/7452c0bf1bea20ff1ba450411de9bf4fa945377c/packages/next/client/image.tsx#L390-L397

Could you try below code:
```js
console.log(typeof product.displayImage)
```
and see what does it output?

**Reply by zhyd1997** [reply]
@gurupal 
@mauricioross 

And could you provide an example that can reproduce your problem?

It's better for debugging the code.

**Reply by simondebbarma** [reply]
Facing this same error when trying to retrieve user image from **next-auth** via `useSession()`. It works with `img` but just refuses to with `Image`. I don't get it.

**Reply by zhyd1997** [reply]
Is there a minimal work example?

**Reply by GabenGar** [reply]
@spdebbarma 
> Facing this same error when trying to retrieve user image from **next-auth** via `useSession()`. It works with `img` but just refuses to with `Image`. I don't get it.

`Image` component is used for the images available at build time. Without additional fiddling with configs that means only files imported as modules and the files located inside `public` folder.

**Reply by ntcaetano** [reply]
Just put <img .... /> and write it in the first line of code:  /* eslint-disable @next/next/no-img-element */

**Reply by martinemahirwe** [reply]
> Facing this same error when trying to retrieve user image from **next-auth** via `useSession()`. It works with `img` but just refuses to with `Image`. I don't get it.

it works for me too. using **Image** is not gonna work when using session. better use the **img** as notmal

**Comment by rahadur** [comment]
I solve this issue by conditionally display `Image` component

```javascript
{ product.displayImage ?
      <Image src={product.displayImage} width="300" height="500"  /> :
      <Image src={'/placeholder.png'}  width="300" height="500"   />
 }
```

**Reply by MathiasKandelborg** [reply]
That's not a solution, it's a workaround.

It doesn't solve the error, it doesn't do what it's supposed to, and it increases performance costs without any real benefit.

**Reply by sulemanelahi** [reply]
@MathiasKandelborg did you findout any better solution?

**Comment by 0xrishavsharma** [comment]
I am also facing this issue. Sometimes the error shows up and sometime it doesn't. 

I saw that when I was normally logging the property it was getting logged in the console properly.
But when I was using it in the src prop, Next JS was throwing me this error.

I don't know why this is happening 🥲

**Comment by ErickCSS** [comment]
### I solve this issue with a conditionally like this: 

`
{ product.displayImage && (
<Image src={product.displayImage} width="300" height="500"  />
)
}
`

I hope i've helped you

**Reply by aldyazarya** [reply]
it really help

**Reply by codesafdar** [reply]
> ### I solve this issue with a conditionally like this:
> `{ product.displayImage && ( <Image src={product.displayImage} width="300" height="500" /> ) }`
> 
> I hope i've helped you

Thanks ,it helped me resolving my issue

**Reply by cosmicknox** [reply]
@ErickCSS thanks this worked for me :)

**Reply by cptkuk91** [reply]
Thank you solved it :)

**Reply by ELHart05** [reply]
Thanks bud it helped

**Reply by boredavinci** [reply]
Solved it for me as well -  just curious - why does it do that? thx!

**Reply by GabenGar** [reply]
@boredavinci 
Because it only passes the props when they are present.

**Reply by sahilbishnoi156** [reply]
it happens because your data is coming late so there is not 'src' value on that point.
You can also solve it like : (ensure that you must have a imageurl)
<Image src={product.displayImage || '/anyimagepath'} width="300" height="500"  /> ) 
or you can use vercel svg logo it doesn't matter because it is just going to display for very small amount of time

**Reply by mohamed-farid-roushdy** [reply]
Thank you solved it

**Reply by natt-marketrue** [reply]
@sahilbishnoi156 your explanation help so much. I was so focus on error with size properties, and couldn't figure this out.

**Reply by sanmazum** [reply]
@sahilbishnoi156 , I applied the same logic but the image is still not getting loaded in the UI

**Reply by sahilbishnoi156** [reply]
@sanmazum I hope you have solved the problem to this point, but if anyone else having the same problem; You have to first import the image and then pass it in src attribute
`import placeholderImg from "./yourimagepath";
<Image src={product.displayImage || placeholderImg} width="300" height="500" />`

**Comment by LuchoMorla** [comment]
I have the same issue, i solved adding the operator AND:
		{ product?.image &&
		<Image src={product?.image} width={300} height={300} alt={product?.name} className={styles.image}/>
		}

**Comment by hamedor** [comment]
I have th same issue, but in my case i needed to add this in next.config.js:
```
images: {
    loader: "default",
    domains: ["localhost"],
  },
```
Then i added absolute path in Image src:

```
<Image
      src={`http://localhost:1337${product.displayImage}`}
 />
```

**Comment by ninja-** [comment]
Just had another non-mentioned case to debug.
Looks like it was a leftover of migration from HTML <img I haven't noticed 👀 

WRONG:
```js
<Image
      src={myStaticImage.src}
 />
```

FIXED:
```js
<Image
      src={myStaticImage}
 />
```

**Comment by synasapmob** [comment]
I think you should do with 'template literals'

<Image src={'${product.displayImage}'} /> symbols ' assume for `

**Reply by inezafrica** [reply]
Thanks so much...It worked form me

**Comment by amirsky799** [comment]
**I had the same issue**
![f1](https://github.com/vercel/next.js/assets/41463587/61702b13-73a1-442d-b10c-e65c6c071bff)

**Section Component**
![f2](https://github.com/vercel/next.js/assets/41463587/835a2af9-4851-43ed-a1f5-bdfafb6763d4)

**My Image Directory** 
![f3](https://github.com/vercel/next.js/assets/41463587/98621780-c231-43ea-b245-93e4dbd10151)

**And I got this error**
![f4](https://github.com/vercel/next.js/assets/41463587/ba5ceb7f-829f-4585-a88d-6c1514e06685)

Any Help ??

**Reply by synasapmob** [reply]
imageLaptop wrong path `public/images/....`

**Reply by sahilbishnoi156** [reply]
if your images are in public folder you don't have to write public just use / for root access.

**Reply by synasapmob** [reply]
sorry this is mistake when answered this question
one more time, i see picture above him used **import** directly then passed **imageLaptop** as an props so you can check with ``console.log(imageLaptop)`` you will recevied an **Module** now i will show you some examples
```jsx
{
    blurDataURL: string
    src: string
    etc....
}
```

good thing right?, thereforce modify them by ``image={{ src: imageLaptop.src }}``

**Comment by sahilbishnoi156** [comment]
This worked for me:
`<Image
        src={routeUser?.image || '/any path you have'}
 />`

**Comment by BoronenkoTimurs** [comment]
Hello, 
Wanted to ask help with my case please. I'm using .map methode to show all images which I applied. But for some reason its show me error: `Image is missing required "src" property:` if I used just src{url}.
When I used `src={ '/loading.jpg' || url}` I try to applied 3 images, but got only one image with loading.png. 

Thank you for help!

Code:
```
'use client';

import Image from 'next/image';
import { ChangeEvent, useEffect, useRef, useState } from 'react';

import Icon from '@/components/custom/icon/components/icon';
import { Button } from '@/components/ui/button';

interface ImageUploadFieldProps {
  disabled?: boolean;
  onChange: (value: string[]) => void;
  onRemove: (value: string) => void;
  value: string[];
};

const ImageUploadField: React.FC<ImageUploadFieldProps> = ({
  disabled,
  onChange,
  onRemove,
  value,
}) => {
  const [mounted, setMounted] = useState(false);
  const fileInputRef = useRef<HTMLInputElement | null>(null);

  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) {
    return null;
  }

  const onUpload = () => {
    if (fileInputRef.current) {
      fileInputRef.current.click();
    }
  };

  const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
    const files = event.target.files;

    if (files) {
      const fileArray = Array.from(files);
      const newUrls: string[] = [];

      fileArray.forEach((file) => {
        const reader = new FileReader();

        reader.onloadend = () => {
          if (reader.result) {
            newUrls.push(reader.result as string);
            console.log('loh1');
            if (newUrls.length === fileArray.length) {
              console.log('loh2');
              onChange([...value, ...newUrls]);
            }
          }
        };

        reader.readAsDataURL(file);
      });
    }
  };

  return (
    <div>
      <div className="mb-4 flex items-center gap-4">
        {value.map((url) => (
          <div 
            key={url} 
            className="relative h-52 w-52 rounded-md overflow-hidden"
          >
            <div className="z-10 absolute top-2 right-2">
              <Button
                type="button"
                variant="destructive"
                size="icon"
                onClick={() => onRemove(url)}
              >
                <Icon name="trash"/>
              </Button>
            </div>
            <Image
              fill
              src={ '/loading.jpg' || url}
              className="object-contain"
              alt="Image"
            />
          </div>
        ))}
      </div>
      <input
        multiple
        type="file"
        ref={fileInputRef}
        disabled={disabled}
        className="hidden"
        onChange={handleFileChange}
      />
      <Button
        type="button"
        variant="secondary"
        disabled={disabled}
        onClick={onUpload}
      >
        <Icon name="plus" className="mr-2"/>
        Upload an Image
      </Button>
    </div>
  );
};

export default ImageUploadField;
```

**Reply by ErickCSS** [reply]
The url, have another parameter?, Because that will be the problem.

Example <Image src={"/loading.png" || url.image} />

**Reply by synasapmob** [reply]
> Hello, Wanted to ask help with my case please. I'm using .map methode to show all images which I applied. But for some reason its show me error: `Image is missing required "src" property:` if I used just src{url}. When I used `src={ '/loading.jpg' || url}` I try to applied 3 images, but got only one image with loading.png.
> 
> Thank you for help!
> 
> Code:
> 
> ```
> 'use client';
> 
> import Image from 'next/image';
> import { ChangeEvent, useEffect, useRef, useState } from 'react';
> 
> import Icon from '@/components/custom/icon/components/icon';
> import { Button } from '@/components/ui/button';
> 
> interface ImageUploadFieldProps {
>   disabled?: boolean;
>   onChange: (value: string[]) => void;
>   onRemove: (value: string) => void;
>   value: string[];
> };
> 
> const ImageUploadField: React.FC<ImageUploadFieldProps> = ({
>   disabled,
>   onChange,
>   onRemove,
>   value,
> }) => {
>   const [mounted, setMounted] = useState(false);
>   const fileInputRef = useRef<HTMLInputElement | null>(null);
> 
>   useEffect(() => {
>     setMounted(true);
>   }, []);
> 
>   if (!mounted) {
>     return null;
>   }
> 
>   const onUpload = () => {
>     if (fileInputRef.current) {
>       fileInputRef.current.click();
>     }
>   };
> 
>   const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
>     const files = event.target.files;
> 
>     if (files) {
>       const fileArray = Array.from(files);
>       const newUrls: string[] = [];
> 
>       fileArray.forEach((file) => {
>         const reader = new FileReader();
> 
>         reader.onloadend = () => {
>           if (reader.result) {
>             newUrls.push(reader.result as string);
>             console.log('loh1');
>             if (newUrls.length === fileArray.length) {
>               console.log('loh2');
>               onChange([...value, ...newUrls]);
>             }
>           }
>         };
> 
>         reader.readAsDataURL(file);
>       });
>     }
>   };
> 
>   return (
>     <div>
>       <div className="mb-4 flex items-center gap-4">
>         {value.map((url) => (
>           <div 
>             key={url} 
>             className="relative h-52 w-52 rounded-md overflow-hidden"
>           >
>             <div className="z-10 absolute top-2 right-2">
>               <Button
>                 type="button"
>                 variant="destructive"
>                 size="icon"
>                 onClick={() => onRemove(url)}
>               >
>                 <Icon name="trash"/>
>               </Button>
>             </div>
>             <Image
>               fill
>               src={ '/loading.jpg' || url}
>               className="object-contain"
>               alt="Image"
>             />
>           </div>
>         ))}
>       </div>
>       <input
>         multiple
>         type="file"
>         ref={fileInputRef}
>         disabled={disabled}
>         className="hidden"
>         onChange={handleFileChange}
>       />
>       <Button
>         type="button"
>         variant="secondary"
>         disabled={disabled}
>         onClick={onUpload}
>       >
>         <Icon name="plus" className="mr-2"/>
>         Upload an Image
>       </Button>
>     </div>
>   );
> };
> 
> export default ImageUploadField;
> ```

are you sure field of ``value.map`` not missing another URL?, can you print console them right now?

**Comment by MarlonEngFront** [comment]
> I solved my case like this:
> `import React, { useState, useEffect } from "react";
> import Image from 'next/image';
> import imageFallback from "../../public/img/portrait_not_found.png";
> 
> interface CarouselItemProps {
>   imageUrl: string | null;
>   title: string;
> }
> 
> const CarolselItemNext: React.FC<CarouselItemProps> = ({ imageUrl, title }) => {
>   const [imgSrc, setImgSrc] = useState<string>(imageFallback.src);
>   const [isLoading, setIsLoading] = useState(true); // Estado para controlar o blur
> 
>   useEffect(() => {
>     if (imageUrl && imageUrl !== 'null') {
>       setImgSrc(imageUrl);
>     } else {
>       setImgSrc(imageFallback.src);
>     }
>   }, [imageUrl]);
> 
>   return (
>     <div className="carousel-item w-40 mx-2">
>       <div className="relative w-full h-60 aspect-[2/3] overflow-hidden">
>         <Image
>           src={imgSrc}
>           alt={title}
>           layout="fill"
>           objectFit="cover"
>           className={isLoading ? 'filter-blur' : 'filter-none'} // Aplica o blur enquanto carrega
>           onLoadingComplete={() => setIsLoading(false)} // Remove o blur quando o carregamento termina
>           onError={() => setImgSrc(imageFallback.src)} // Define a imagem de fallback em caso de erro
>           placeholder="blur"
>           blurDataURL={imageFallback.src}
>           priority={false}
>           loading="lazy"
>         />
>       </div>
>       <p className="text-center text-white mt-2">{title}</p>
>     </div>
>   );
> };
> 
> export default CarolselItemNext;
> `

**Comment by playstore777** [comment]
Thanks to Copilot, issue resolved.

I am not sure what is the exact fix, I tried to add null check but that didn't solve.
But Copilot fixed it, and the only thing which was added different was layout="fill" and objectFit="cover"

```
{imageURL && (
            <Image
              src={imageURL}
              alt="random-person-image"
              className={classes.image}
              layout="fill"
              objectFit="cover"
            />
          )}

```
It works

the error I got 
![image](https://github.com/user-attachments/assets/c68296c4-9ff3-44fc-805c-79afeb7b0f41)

Nope adding "width" etc didn't solve, but somehow the above code solved it!

**Reply by sahilbishnoi156** [reply]
Image component in next.js expect exact resolution of image for efficient rendering. So either you provide height and width attribute or you provide the layout attribute with position set to "relative" of parent component of Image Component. And their is no special or must use objectFit attribute it is just a style for image. Hope that helps
